### PR TITLE
Add bracket

### PR DESCRIPTION
**Pull request by rightfold**

### CODE REVIEW

**Comment by paf31** [comment]
Looks good, but I wonder if there's a better name. I know Haskell uses `bracket`, but it's not great IMO. What about `withResource`?

**Comment by rightfold** [comment]
Yeah perhaps 😂 I like that name.

Sent from my iPhone.

**Comment by paf31** [comment]
👍 Looks good to me, thanks!

File: src/Control/Monad/Error/Class.purs [hunk 1]
```diff
@@ -54,3 +54,19 @@ instance monadErrorMaybe :: MonadError Unit Maybe where
   throwError = const Nothing
   catchError Nothing f  = f unit
   catchError (Just a) _ = Just a
+
+-- | Make sure that a resource is cleaned up in the event of an exception. The
+-- | release action is called regardless of whether the body action throws or
+-- | returns.
+withResource
+  :: ∀ e m r a
```
Comment (paf31) [inline src/Control/Monad/Error/Class.purs]: Can we use `forall` here?

File: src/Control/Monad/Error/Class.purs [hunk 2]
```diff
@@ -54,3 +54,19 @@ instance monadErrorMaybe :: MonadError Unit Maybe where
...
+   . (MonadError e m)
```
Comment (paf31) [inline src/Control/Monad/Error/Class.purs]: Redundant parens

**Comment by rightfold** [comment]
@paf31 Any more comments on this?

**Comment by paf31** [comment]
@rightfold It looks good, thanks. I have a mountain of PRs to merge, but we'll get them merged before the next major releases (which will coincide with 0.11).

**Comment by paf31** [comment]
Actually, no reason this can't go in now.

**Pull request merged**